airflow/www/forms.py

```diff
@@ -276,71 +230,14 @@ class ConnectionForm(DynamicForm):
-    )
-    extra__google_cloud_platform__keyfile_dict = PasswordField(
-        lazy_gettext('Keyfile JSON'), widget=BS3PasswordFieldWidget()
-    )
-    extra__google_cloud_platform__scope = StringField(
-        lazy_gettext('Scopes (comma separated)'), widget=BS3TextFieldWidget()
-    )
-    extra__google_cloud_platform__num_retries = IntegerField(
-        lazy_gettext('Number of Retries'),
-        validators=[NumberRange(min=0)],
-        widget=BS3TextFieldWidget(),
-        default=5,
-    )
-    extra__grpc__auth_type = StringField(lazy_gettext('Grpc Auth Type'), widget=BS3TextFieldWidget())
-    extra__grpc__credential_pem_file = StringField(
-        lazy_gettext('Credential Keyfile Path'), widget=BS3TextFieldWidget()
-    )
-    extra__grpc__scopes = StringField(lazy_gettext('Scopes (comma separated)'), widget=BS3TextFieldWidget())
-    extra__yandexcloud__service_account_json = PasswordField(
-        lazy_gettext('Service account auth JSON'),
-        widget=BS3PasswordFieldWidget(),
-        description='Service account auth JSON. Looks like '
-        '{"id", "...", "service_account_id": "...", "private_key": "..."}. '
-        'Will be used instead of OAuth token and SA JSON file path field if specified.',
-    )
-    extra__yandexcloud__service_account_json_path = StringField(
-        lazy_gettext('Service account auth JSON file path'),
-        widget=BS3TextFieldWidget(),
-        description='Service account auth JSON file path. File content looks like '
-        '{"id", "...", "service_account_id": "...", "private_key": "..."}. '
-        'Will be used instead of OAuth token if specified.',
-    )
-    extra__yandexcloud__oauth = PasswordField(
-        lazy_gettext('OAuth Token'),
-        widget=BS3PasswordFieldWidget(),
-        description='User account OAuth token. Either this or service account JSON must be specified.',
-    )
-    extra__yandexcloud__folder_id = StringField(
-        lazy_gettext('Default folder ID'),
-        widget=BS3TextFieldWidget(),
-        description='Optional. This folder will be used to create all new clusters and nodes by default',
-    )
-    extra__yandexcloud__public_ssh_key = StringField(
-        lazy_gettext('Public SSH key'),
-        widget=BS3TextFieldWidget(),
-        description='Optional. This key will be placed to all created Compute nodes'
-        'to let you have a root shell there',
-    )
-    extra__kubernetes__in_cluster = BooleanField(lazy_gettext('In cluster configuration'))
-    extra__kubernetes__kube_config_path = StringField(
-        lazy_gettext('Kube config path'), widget=BS3TextFieldWidget()
-    )
-    extra__kubernetes__kube_config = StringField(
-        lazy_gettext('Kube config (JSON format)'), widget=BS3TextFieldWidget()
-    )
-    extra__kubernetes__namespace = StringField(lazy_gettext('Namespace'), widget=BS3TextFieldWidget())
+
+
+for key, value in ProvidersManager().connection_form_widgets.items():
+    setattr(ConnectionForm, key, value.field)
```

Comment:
This causes plugins to be loaded at webserver boot -- we should delay this to as late as possible. There is a config option to control eager loading or not of plugins. With this code, that has no effect anymore as this will always eager load it.

Reply:
Any suggestion at which time/where to add it to be better? I Am not 100% sure how/when FAB instantiates the classes, But we could definitely do it at other convenient place. There are just two instantiations thate are problematic:

form:ConnectionForm
views: ConnectionModelView

We could create ProvidersManager in ConnectionModelView class definition, but that will also be executed when view.py is imported (but maybe it is imported for the first time when FAB is trying to display anything for the first time?)

Reply:
I think I found a perfect place and I also do the field cusomisation in the same PR as they seem to be very nicely connected now.